ecosystem/sf-indexer/aptos-sf-indexer/migrations/2022-08-08-043603_core_tables/up.sql

```diff
@@ -0,0 +1,285 @@
+ "data":{
+ "amazing":"2"
+ }
+ }
+ ],
+ "timestamp":"1649713141723410"
+ }
+ */
+CREATE TABLE user_transactions (
+  version BIGINT UNIQUE PRIMARY KEY NOT NULL,
+  block_height BIGINT NOT NULL,
+  parent_signature_type VARCHAR(255) NOT NULL,
+  sender VARCHAR(255) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  max_gas_amount NUMERIC NOT NULL,
+  expiration_timestamp_secs TIMESTAMP NOT NULL,
+  gas_unit_price NUMERIC NOT NULL,
+  -- from UserTransaction
+  "timestamp" TIMESTAMP NOT NULL,
+  -- Default time columns
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  CONSTRAINT fk_versions FOREIGN KEY (version) REFERENCES transactions (version),
+  UNIQUE (sender, sequence_number)
+);
+CREATE INDEX ut_sender_index ON user_transactions (sender);
+CREATE TABLE signatures (
+  transaction_version BIGINT NOT NULL,
+  multi_agent_index BIGINT NOT NULL,
+  multi_sig_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  signer VARCHAR(255) NOT NULL,
+  is_sender_primary BOOLEAN NOT NULL,
+  type VARCHAR(255) NOT NULL,
+  public_key VARCHAR(255) NOT NULL,
+  threshold BIGINT NOT NULL,
+  public_key_indices jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (
+    transaction_version,
+    multi_agent_index,
+    multi_sig_index
+  ),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+/** Ex:
+ {
+ "key": "0x0400000000000000000000000000000000000000000000000000000000000000000000000a550c18",
+ "sequence_number": "0",
+ "type": "0x1::reconfiguration::NewEpochEvent",
+ "data": {
+ "epoch": "1"
+ }
+ }
+ */
+CREATE TABLE events (
+  key VARCHAR(100) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  creation_number BIGINT NOT NULL,
+  account_address VARCHAR(100) NOT NULL,
+  transaction_version BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  data jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (key, sequence_number),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX event_key_seq_type_index ON events (key, sequence_number, type);
+CREATE TABLE write_set_changes (
+  transaction_version BIGINT NOT NULL,
+  hash VARCHAR(255) NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  address VARCHAR(255) NOT NULL,
+  index BIGINT NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (transaction_version, hash),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX write_set_changes_tx_hash_addr_type_index ON write_set_changes (transaction_version, address, type);
+CREATE TABLE move_modules (
+  transaction_version BIGINT NOT NULL,
+  write_set_change_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
```

Comment:
if we have `transaction_block_height` everywhere; do we also want to throw an index on it?

Reply:
I suspect the query pattern will still be on version tho, especially if there are thousands of versions in a block. Frankly rn I have no idea what the query pattern will be.